apps/nextjs-website/src/components/PressReleaseList.tsx

```diff
@@ -14,28 +31,44 @@ const makePressReleaseListProps = ({
 }: PressReleaseListSection &
   SiteWidePageData & {
     pressReleasePages: ReadonlyArray<PressReleasePage>;
-  }): PressReleaseListProps => ({
-  ...rest,
-  pressReleases: pressReleasePages.map((PRPage) => ({
-    date: PRPage.pressRelease.date,
-    title: PRPage.pressRelease.title,
-    link: {
-      label: locale === 'it' ? 'Leggi' : 'Read',
-      href: LocalizeURL({
-        URL: '/press-releases/' + PRPage.slug,
-        locale,
-        defaultLocale,
-      }),
-    },
-    themeVariant: rest.themeVariant,
-  })),
-});
+  }): PressReleaseListProps => {
+  const resolvedLocale = locale || defaultLocale || 'en-GB';
+
+  return {
+    ...rest,
+    locale: resolvedLocale,
+    pressReleases: pressReleasePages.map((PRPage) => ({
+      date: formatDateToLocale(PRPage.pressRelease.date, resolvedLocale),
+      title: PRPage.pressRelease.title,
+      link: {
+        label: resolvedLocale.startsWith('it') ? 'Leggi' : 'Read',
+        href: LocalizeURL({
+          URL: '/press-releases/' + PRPage.slug,
+          locale: resolvedLocale,
+          defaultLocale,
+        }),
+      },
+      themeVariant: rest.themeVariant,
+      locale: resolvedLocale,
+    })),
+  };
+};
```

Comment:
```suggestion
  }): PressReleaseListProps => ({
    ...rest,
    pressReleases: pressReleasePages.map((PRPage) => ({
      date: formatDateToLocale(PRPage.pressRelease.date, locale),
      title: PRPage.pressRelease.title,
      link: {
        label: locale === 'it' ? 'Leggi' : 'Read',
        href: LocalizeURL({
          URL: '/press-releases/' + PRPage.slug,
          locale,
          defaultLocale,
        }),
      },
      themeVariant: rest.themeVariant,
    })),
  });
};
```
Just use locale. Types make sure it has a value.